src/contracts/examples/redstone-oev/RedstoneOevDappControl.sol

```diff
@@ -93,54 +124,105 @@ contract RedstoneOevDappControl is DAppControl {
+        // The bundler must be whitelisted
+        RedstoneOevDappControl(_control()).verifyBundlerWhitelist();
+
+        // The user must be calling the present contract
+        if (userOp.dapp != _control()) revert InvalidUserDestination();
 
-        address oracle = abi.decode(returnData, (address));
+        // The user must be calling the update function
+        if (bytes4(userOp.data) != bytes4(RedstoneOevDappControl.update.selector)) {
+            revert InvalidUserEntryCall();
+        }
+
+        (address _oracle, bytes memory _updateCallData) = abi.decode(userOp.data[4:], (address, bytes));
 
-        uint256 oevPercentBundler = RedstoneOevDappControl(_control()).bundlerOEVPercent();
-        uint256 oevPercentAtlas = RedstoneOevDappControl(_control()).atlasOEVPercent();
-        address vaultAtlas = RedstoneOevDappControl(_control()).atlasVault();
-        address vaultOracle = RedstoneOevDappControl(_control()).oracleVaults(oracle);
+        // The called oracle must be whitelisted
+        RedstoneOevDappControl(_control()).verifyOracleWhitelist(_oracle);
 
-        uint256 bundlerOev = (bidAmount * oevPercentBundler) / 100;
-        uint256 atlasOev = (bidAmount * oevPercentAtlas) / 100;
-        uint256 oracleOev = bidAmount - bundlerOev - atlasOev;
+        // The update call data must be a valid updateDataFeedsValues call
+        if (bytes4(_updateCallData) != bytes4(IRedstoneAdapter.updateDataFeedsValues.selector)) {
+            revert InvalidUserUpdateCall();
+        }
 
-        if (oracleOev > 0) SafeTransferLib.safeTransferETH(vaultOracle, oracleOev);
-        if (atlasOev > 0) SafeTransferLib.safeTransferETH(vaultAtlas, atlasOev);
-        if (bundlerOev > 0) SafeTransferLib.safeTransferETH(_bundler(), bundlerOev);
+        return "";
+    }
+
+    /**
+     * @notice Allocates the bid amount to the relevant parties
+     * @param bidAmount The bid amount to be allocated
+     * @dev This function is delegatcalled
+     */
+    function _allocateValueCall(address, uint256 bidAmount, bytes calldata) internal virtual override {
+        if (bidAmount == 0) return;
+
+        // Get the OEV share for the bundler and transfer it
+        uint256 _oevShareBundler = bidAmount * RedstoneOevDappControl(_control()).oevShareBundler() / OEV_SHARE_SCALE;
+        if (_oevShareBundler > 0) SafeTransferLib.safeTransferETH(_bundler(), _oevShareBundler);
+
+        // Get the OEV share for Fastlane and transfer it
+        uint256 _oevShareFastlane = bidAmount * RedstoneOevDappControl(_control()).oevShareFastlane() / OEV_SHARE_SCALE;
+        if (_oevShareFastlane > 0) SafeTransferLib.safeTransferETH(oevAllocationDestination, _oevShareFastlane);
+
+        // Transfer the rest
+        SafeTransferLib.safeTransferETH(
+            RedstoneOevDappControl(_control()).oevAllocationDestination(),
+            bidAmount - _oevShareBundler - _oevShareFastlane
+        );
```

Comment:
Maybe we should calculate the result of `bidAmount - _oevShareBundler - _oevShareFastlane` and only if not zero, do the transfer call. Looks like it could be zero from the restrictions on the setter functions.

Reply:
Done in https://github.com/FastLane-Labs/atlas/pull/435/commits/c47999f6eade718b777933ed354e26004a1773ad.